openpype/hosts/maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -83,3 +95,33 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
         self.log.info("Extracted instance '%s' to: %s" % (instance.name, path))
+
+    @staticmethod
+    def _add_loaded_containers(members):
+        # type: (list) -> list
+        refs_to_include = [
+            cmds.referenceQuery(ref, referenceNode=True)
+            for ref in members
+            if cmds.referenceQuery(ref, isNodeReferenced=True)
+        ]
+
+        refs_to_include = set(refs_to_include)
+
+        obj_sets = cmds.ls("*.id", long=True, type="objectSet", recursive=True,
+                           objectsOnly=True)
+
+        loaded_containers = []
+        for obj_set in obj_sets:
+
+            if not cmds.attributeQuery("id", node=obj_set, exists=True):
+                continue
+
+            id_attr = "{}.id".format(obj_set)
+            if cmds.getAttr(id_attr) != AVALON_CONTAINER_ID:
+                continue
+
+            set_content = set(cmds.sets(obj_set, query=True))
+            if set_content.intersection(refs_to_include):
+                loaded_containers.append(obj_set)
+
+        return loaded_containers
```

Comment:
I think this logic would fail to detect e.g. loaded `gpuCache` containers, `yetiCache` containers, etc. and others that aren't referenced. So this logic would not be correct?

```python
members_with_refs= set(members) + set(refs_to_include)
...

if set_content.intersection(members_with_refs)
```

Likely you'd need something like that?